### ISSUE TITLE
azuredevops_git_permissions fails when adding team

### ISSUE THREAD

**Issue by irogers97** [microsoft/terraform-provider-azuredevops#526]
### Terraform (and Azure DevOps Provider) Version

Terraform v1.1.4
on darwin_amd64

### Affected Resource(s)

azuredevops_git_permissions 

### Terraform Configuration Files
```hcl
data "azuredevops_project" "this" {
  name = var.ado_project_name
}

data "azuredevops_git_repository" "this" {
  project_id = data.azuredevops_project.this.id
  name       = var.repository_name
}

data "azuredevops_team" "this" {
  for_each   = toset(var.repository_teams)
  project_id = data.azuredevops_project.this.id
  name       = each.value
}

resource "azuredevops_git_permissions" "this" {
  for_each      = toset(values(data.azuredevops_team.this)[*].id)
  project_id    = data.azuredevops_project.this.id
  repository_id = data.azuredevops_git_repository.this.id
  principal     = each.value
  permissions   = var.repository_permissions
}
```
### Debug Output
```log
2022-01-25T11:54:40.886Z [TRACE] Re-validating config for "module.ado_git_permissions_administrators.azuredevops_git_permissions.this[\"b37bdc2a-021c-4be7-a136-36a90a276232\"]"
2022-01-25T11:54:40.886Z [TRACE] GRPCProvider: ValidateResourceConfig
2022-01-25T11:54:40.887Z [TRACE] GRPCProvider: PlanResourceChange
2022-01-25T11:54:40.888Z [WARN]  Provider "registry.terraform.io/microsoft/azuredevops" produced an invalid plan for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"], but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .replace: planned value cty.True for a non-computed attribute
2022-01-25T11:54:40.888Z [TRACE] checkPlannedChange: Verifying that actual change (action Create) matches planned change (action Create)
2022-01-25T11:54:40.888Z [TRACE] readResourceInstanceState: reading state for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:40.888Z [TRACE] readResourceInstanceState: no state present for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:40.888Z [INFO]  Starting apply for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:40.889Z [DEBUG] module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]: applying the planned Create change
2022-01-25T11:54:40.889Z [TRACE] GRPCProvider: ApplyResourceChange
2022-01-25T11:54:41.240Z [TRACE] maybeTainted: module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"] encountered an error during creation, so it is now marked as tainted
2022-01-25T11:54:41.240Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:41.240Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: removing state object for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:41.240Z [TRACE] evalApplyProvisioners: module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"] is tainted, so skipping provisioning
2022-01-25T11:54:41.240Z [TRACE] maybeTainted: module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"] was already tainted, so nothing to do
2022-01-25T11:54:41.240Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:41.240Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: removing state object for module.ado_git_permissions_administrators.azuredevops_git_permissions.this["b37bdc2a-021c-4be7-a136-36a90a276232"]
2022-01-25T11:54:41.240Z [ERROR] vertex "module.ado_git_permissions_administrators.azuredevops_git_permissions.this[\"b37bdc2a-021c-4be7-a136-36a90a276232\"]" error: The string must have at least one character.^M
Parameter name: descriptors element.IdentityType
2022-01-25T11:54:41.240Z [TRACE] vertex "module.ado_git_permissions_administrators.azuredevops_git_permissions.this[\"b37bdc2a-021c-4be7-a136-36a90a276232\"]": visit complete, with errors
2022-01-25T11:54:41.240Z [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/microsoft/azuredevops\"] (close)" errored, so skipping
2022-01-25T11:54:41.240Z [TRACE] dag/walk: upstream of "module.ado_git_permissions_administrators (close)" errored, so skipping
2022-01-25T11:54:41.240Z [TRACE] dag/walk: upstream of "root" errored, so skipping
```
### Panic Output

None

### Expected Behavior

Team added to repo security

### Actual Behavior

Terraform error:
```log
Error: The string must have at least one character.
│ Parameter name: descriptors element.IdentityType
│
│   with module.ado_git_permissions_developers.azuredevops_git_permissions.this["*****-***-***-******"],
│   on ../../../../resources/azuredevops/git/permissions/_main.tf line 16, in resource "azuredevops_git_permissions" "this":
│   16: resource "azuredevops_git_permissions" "this" {

```
### Steps to Reproduce

1. `terraform apply`

### Important Factoids

n/a

**Comment by xuzhang3** [comment]
@irogers97 `azuredevops_git_permissions.principal` accepts the resource descriptor not an ID. For group resource, the ID is the descriptor but for `azuredevops_team` this the descriptor is not returned by service. API https://docs.microsoft.com/en-us/rest/api/azure/devops/graph/descriptors/get?view=azure-devops-rest-6.0 can be used to resolve the ID to descriptor but this resource is not supported for now.

**Comment by irogers97** [comment]
@xuzhang3 As the `azuredevops_team` resource and data source both returns descriptors of the administrators and members, it would be useful if the resource and data source also returned the descriptor of the team.
Looks to me like the team api, [https://docs.microsoft.com/en-us/rest/api/azure/devops/core/teams/get?view=azure-devops-rest-6.0](https://docs.microsoft.com/en-us/rest/api/azure/devops/core/teams/get?view=azure-devops-rest-6.0), already provides the descriptor

**Comment by xuzhang3** [comment]
@irogers97  Not all teams will return the `descriptor`, but ID will. #528 I will submit a PR add a new resource to support resolve the ID to descriptor.

**Comment by xuzhang3** [comment]
Fixed in #753, Close this issue
